src/main/java/io/fabric8/elasticsearch/plugin/filter/FieldStatsResponseFilter.java

```diff
@@ -72,7 +73,20 @@ public void apply(final Task task, final String action, final ActionRequest requ
             @SuppressWarnings("unchecked")
             @Override
             public void onResponse(ActionResponse response) {
-                listener.onResponse(response);
+                if(INDICES_FIELD_STATS_READ_ACTION.equals(action) && response instanceof FieldStatsResponse) {
+                    if(((FieldStatsResponse)response).getIndicesMergedFieldStats().size() == 0) {
+                        LOGGER.trace("Modifying the response to be {}", RestStatus.NO_CONTENT);
+                        Throwable err = new ElasticsearchException("The index returned an empty result."
+                                + "You can use the Time Picker to change the time filter or select a higher time interval",
+                                RestStatus.NO_CONTENT);
+                        
+                        listener.onFailure(err);
+                    } else {
```

Comment:
Probably instead of the nested if/else, simply utilize an earlier return after 'onfailure'
```
if(FieldStatsResponse){
...
onFailure(err)
return
}
onResponse()
```